### ISSUE TITLE
[Suggestion] [swade] Trigger the animation only on damage and/or spell rolls

### ISSUE THREAD

**Issue by humsi** [otigon/automated-jb2a-animations#516]
As suggested on the #swade Discord channel:
I don’t know if it’s intentional, but using the Savage Worlds/swade system (by FloRadical), animations are played both on the trait/skill and the damage roll. 
My suggestion: 
Extend the already used hook “swadeAction“:
```
Hooks.on("swadeAction", async (SwadeTokenOrActor, SwadeItem) => {...}
```
with
```
Hooks.on("swadeAction", async (SwadeTokenOrActor, SwadeItem, SwadeAction) => {...}
```

And within that block check if the "SwadeAction" is either a damage roll or a skill roll from an item without damage and than trigger the animation, like:
```
if (SwadeAction == "damage" || (SwadeAction == "formula" && SwadeItem.system.damage == "")) {
    runSwade(SwadeTokenOrActor, SwadeItem)
}
```

The hook’s third argument ("SwadeAction") provides information which kind of roll is performed (e.g., formula = trait/skill roll, damage = damage roll).
To explain the dice mechanics: Roll skill to hit (e.g., skill “Shooting” → no animation), if successful: roll damage (e.g. 2d6 → animation). 

However, there are “items” (like spells) without damage, whose animations should be played on skill/trait rolls.
Basically, the animation should be triggered if an item uses a "damage" action, like dealing damage by shooting a gun, striking with a club or casting a fireball. Or if an item without damage uses the "formula" action (e.g., spells like “Healing” or “Teleport” and therefore rolling spell casting skills). The latter animation should be played as you make a skill check using the item. 

I hope my suggestion makes sense and is in principle welcomed by the swade community (maybe someone could check it out to see if there are any unintended side effects?).

Just FYI: The fourth argument passed by the hook is the roll result.

Anyway: Thanks for your great work!

**Comment by otigon** [comment]
How are you supposed to Roll items in SWADE? I've only ever been able to get things to work when using Better Rolls for SWADE
If you can give me a step-by-step on using an item in SWADE that would help

**Comment by otigon** [comment]
This is the primary reason I've always relied on Pull Requests to get anything done in the system... it's very different from any other system I've played and getting anything to work is a headache

**Comment by humsi** [comment]
> How are you supposed to Roll items in SWADE? I've only ever been able to get things to work when using Better Rolls for SWADE If you can give me a step-by-step on using an item in SWADE that would help

An "item" in the swade system is not limited to gear.
It can be a weapon or a spell, or even an edge (like a feat in DnD) or hindrance (like a handicap). 
But you only use the roll mechanics from spells, weapons and general equipment. Therefore, a suitable ability must be defined in the item properties (like shooting, fighting, spellcasting, healing and so on). Ideally, the character also has the respective ability in the character sheet. Additionally, of course, the damage must be defined (here only melee attack, therefore damage=strength)

![screenshot](https://user-images.githubusercontent.com/83549920/197359711-8b4b2487-e4f8-404e-b5d7-91307f875768.png)
![screenshot2](https://user-images.githubusercontent.com/83549920/197359821-ea0aa6ef-f8be-4524-a698-d78a736f8dd1.png)

Based on this, you can then either roll the appropriate trait/skill/ability or directly roll the damage.
Either click on the corresponding buttons or use the chat card that is generated when you click on the small speech bubble in the right corner.
![schreenshot3](https://user-images.githubusercontent.com/83549920/197359903-91b98eae-e25b-4b02-87d3-e0912e826e54.png)


That being said: There is currently no combat automation in the swade system for automatic hit or damage calculation. Therefore, many people use either SWADE Tools or Better Rolls 2 for Swade.

**Comment by otigon** [comment]
Awesome thanks for the explanation and screenshots. I didn't have any trait set on the Weapons I was using, nor damage fields. I should be able to get something working with this
